### ISSUE TITLE
feat(outputs.datadog): Support for Dogstatsd Events

### ISSUE THREAD

**Issue by jdheyburn** [influxdata/telegraf#16130]
### Use Case

Dogstatsd allows for events to be published to a StatsD server (typically Datadog Agent) which then is displayed as an Event in Datadog.

https://docs.datadoghq.com/service_management/events/guides/dogstatsd/

In the below PR, inputs.statsd was modified to support parsing the event datagram into a `telegraf.Metric`. 
- https://github.com/influxdata/telegraf/pull/5791

### Expected behavior

An event is published to Datadog.

### Actual behavior

No event is published to Datadog.

### Additional info

It doesn't look like outputs.datadog supports submitting events. It assumes all data received from the accumulator are metric series and converts them as such. The URL even defaults to `https://app.datadoghq.com/api/v1/series`, whereas events are submitted to [Events API](https://docs.datadoghq.com/api/latest/events/) at `https://api.datadoghq.com/api/v1/events`.

Here is an example event that appears in outputs.datadog:

```json
  {
    "MetricName": "jdheyburn test event",
    "MetricTags": [
      { "Key": "datadog-statsd", "Value": "true" },
      { "Key": "host", "Value": "telegraf-events-0" },
    ],
    "MetricFields": [
      { "Key": "priority", "Value": "normal" },
      { "Key": "alert_type", "Value": "info" },
      { "Key": "text", "Value": "from statsd" }
    ],
    "MetricTime": "2024-11-01T19:59:54.891658849Z",
    "MetricType": 3
  },
```

MetricType is telegraf.Untyped. Currently its trying to convert it to a metric and submit it to the series endpoint, whereas it should convert it to an Event and submit it to the correct endpoint.

**Comment by jdheyburn** [comment]
- Should [convertToDatadogMetric](https://github.com/influxdata/telegraf/blob/5793ff636684a0fd1f53e202aea2e7e4dc07de1e/plugins/outputs/datadog/datadog.go#L141) contain some logic to determine from `metrics` what are series and what are events?
  - From there, it should forward the parsed series and events to their respective endpoint

**Comment by srebhan** [comment]
@jdheyburn how would Telegraf know which metric is an event? There is no information in the metric itself...

**Comment by jdheyburn** [comment]
@srebhan Knowing that a valid metric has this format:

```json
  {
    "MetricName": "datadog.dogstatsd.client.bytes_sent",
    "MetricTags": [
      { "Key": "datadog-statsd", "Value": "true" },
      { "Key": "host", "Value": "telegraf-events-0" },
      { "Key": "metric_type", "Value": "counter" }
    ],
    "MetricFields": [{ "Key": "value", "Value": 1158 }],
    "MetricTime": "2024-11-01T19:59:56Z",
    "MetricType": 1
  },
```

Here are some options:

1. Assume the absence of a `metric_type` (or `value` MetricField) makes this an event
2. Make a change to `inputs.statsd` to add a differentiator, maybe a new MetricTag to indicate its an event?
3. Create a [new ValueType](https://github.com/influxdata/telegraf/blob/5793ff636684a0fd1f53e202aea2e7e4dc07de1e/metric.go#L12) of `Event` and set incoming events in `inputs.statsd` to this type (instead of Untyped)

**Comment by srebhan** [comment]
@jdheyburn well a Telegraf metric might or might not have a type. The metric also might or might not have a field called `value` so those a no criteria to judge if this should be an event **in general**!

Your option 3 might work but is too intrusive as some outputs _rely_ on the type of the metric so introducing a new type means we would need to go through all output plugins and "fix" them if necessary...

This leaves us with defining a "rule" or "filter" in the `statsd` output plugin which, IMO, is the only viable option... It would be super helpful if you could come up with a notion of how this "filter" could/should look like for a general use-case! And of course: PRs are welcome! ;-)

**Comment by jdheyburn** [comment]
@srebhan Thanks for the suggestion.

I've not worked with rules or filters in telegraf before, are you able to expand on this and show some examples? In your reply did you mean `datadog` output plugin, or the `statsd` input plugin?

**Comment by srebhan** [comment]
@jdheyburn sorry I meant the `datadog` output plugin. :-/ With filter I mean one or more `glob` expressions and using the [filter implementation](https://github.com/influxdata/telegraf/blob/master/filter/filter.go). So you add e.g.
```toml
  event_metrics = ["datadog.dogstatsd.client.*", "*.event"]
```
as option and then if the generated datadog metric-name matches the expression, the metric will be sent in the event format. Does that make sense?

**Comment by jdheyburn** [comment]
@srebhan Makes sense, thanks for the explanation. I can't promise I'll have the capacity to implement this. Our workaround has been to forward statsd events to the Datadog agent, while keeping statsd metrics submitted to telegraf.
